### PR TITLE
refactor(admin): stream_routes/upstreams/protos/services/global_rules/consumer_groups/plugin_configs

### LINKED ISSUE #8569: feat: As a user, I want to refactor the resources under apisix.admin, so that we don't need to repeat
### Description

Currently, all the resources under `apisix.admin` are basically copied and pasted from a template and changed. If you make general changes to the resource, such as fix count or add adapter, you need to change it file by file. And due to historical problems, some resource implementation methods are incomplete.
I hope we can refactor the code to something like `core.config.new`, where we only need to pass a few parameters to create a new resource.

Note: Please read every resource file before jumping into the refactor. Some resources have particular logic. We need a smart way to handle them.

CC @An-DJ

### PR DESCRIPTION
**Pull request by An-DJ**
### Description

Fixes #8569 

- [X] stream_routes
- [X] upstreams
- [X] protos
- [X] services
- [X] global_rules
- [X] consumer_groups
- [X] plugin_configs
### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [X] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [X] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/admin/consumer_group.lua [hunk 1]
```diff
@@ -59,54 +34,11 @@ local function check_conf(id, conf, need_id)
...
-    end
-    local res, err = core.etcd.get(key, not id)
-    if not res then
-        core.log.error("failed to get consumer group[", key, "]: ", err)
-        return 503, {error_msg = err}
-    end
-
-    utils.fix_count(res.body, id)
-    return res.status, res.body
+    return need_id and id or true
```
Comment (spacewander) [inline apisix/admin/consumer_group.lua:37]: Look like we can move this line to resource.lua?

File: apisix/admin/routes.lua [hunk 2]
```diff
@@ -143,5 +143,6 @@ return resource.new({
     name = "routes",
     kind = "route",
     schema = core.schema.route,
-    checker = check_conf
+    checker = check_conf,
+    unsupported_methods = {}
```
Comment (spacewander) [inline apisix/admin/routes.lua:147]: We can leave this nil as this resource doesn't have unsupported methods.

File: apisix/admin/services.lua [hunk 3]
```diff
@@ -189,75 +99,15 @@ function _M.delete(id)
...
 end
 
 
-return _M
+return resource.new({
+    name = "services",
+    kind = "service",
+    schema = core.schema.service,
+    checker = check_conf,
+    unsupported_methods = {},
```
Comment (spacewander) [inline apisix/admin/services.lua:111]: Ditto

File: apisix/admin/resource.lua [hunk 4]
```diff
@@ -57,7 +57,13 @@ function _M:check_conf(id, conf, need_id)
     core.log.info("conf  : ", core.json.delay_encode(conf))
 
     -- check the resource own rules
-    return self.checker(id, conf, need_id, self.schema)
+    local ok, err = self.checker(id, conf, need_id, self.schema)
+
+    if err then
```
Comment (spacewander) [inline apisix/admin/resource.lua:62]: Code style, please use `if not ok then`

**Review by spacewander — APPROVED** [review]

File: apisix/admin/init.lua [hunk 5]
```diff
@@ -199,7 +199,17 @@ local function run()
     end
 
     local code, data
-    if seg_res == "routes" then
+    local refactored_resources = {
```
Comment (soulbird) [inline apisix/admin/init.lua:202]: directly named as `resources`?

File: apisix/admin/resource.lua [hunk 6]
```diff
@@ -57,11 +57,21 @@ function _M:check_conf(id, conf, need_id)
...
+    if not ok then
+        return ok, err
+    else
+        return need_id and id or true
+    end
 end
 
 
 function _M:get(id)
+    if self.unsupported_methods and core.table.array_find(self.unsupported_methods, "get") then
```
Comment (soulbird) [inline apisix/admin/resource.lua:71]: Is it enough to use `if core.table.array_find(self.unsupported_methods, "get") then` ?

File: apisix/admin/init.lua (hunk 5 above)
Comment (An-DJ) [reply apisix/admin/init.lua:202]: There has been a global variable named `resources` 😭. Maybe we can rename that variable after all resources are refactored.

File: apisix/admin/resource.lua (hunk 6 above)
Comment (An-DJ) [reply apisix/admin/resource.lua]: @soulbird Done.

File: apisix/admin/init.lua (hunk 5 above)
Comment (soulbird) [reply apisix/admin/init.lua:202]: ok

**Review by soulbird — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**